src/tests/JIT/Regression/JitBlue/Runtime_58874/Runtime_58874.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+unsafe class Runtime_58874
+{
+    private static int Main(string[] args)
+    {
+        void* mem;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            const int MEM_COMMIT = 0x1000;
+            const int MEM_RESERVE = 0x2000;
+            const int PAGE_READWRITE = 0x04;
+
+            // Reserve 2 pages
+            void* pages = VirtualAlloc(null, 0x2000, MEM_RESERVE, PAGE_READWRITE);
+            if (pages == null)
+            {
+                return 100;
+            }
+            // Commit first page
+            mem = VirtualAlloc(pages, 0x1000, MEM_COMMIT, PAGE_READWRITE);
+            if (mem != pages)
+            {
+                return 100;
+            }
+        }
+        else
+        {
+            mem = NativeMemory.Alloc(0x1000);
```

Comment:
just curious - is this enough to reproduce the issue on non-windows platforms?

Reply:
No, this will not guarantee repro for non-Windows. But given the JIT code with the bug is platform agnostic I think it is ok to only have guaranteed coverage on Windows instead of special casing the test for more platforms.

Reply:
I agree, was just curious if there an easy way to reproduce it on linux, in theory we can use MemoryMappedFile API

Reply:
Hmm, looks like `MemoryMappedFile` commits the entire range immediately, but we could probably increase our changes by using it.
I suppose the "really correct way" would be to pinvoke `mmap` and `mprotect` on Linux platforms.